### ISSUE TITLE
Graph not working

### ISSUE THREAD

**Issue by bastianvs** [linkedin/kafka-monitor#9]
I am able to run the test but the graphs are not ploting no error in the logs.

**Comment by lindong28** [comment]
@bastianvs Are you able to view the data in the console or query metric using HTTP request, if you start Kafka Monitor using `./bin/kafka-monitor-start.sh config/kafka-monitor.properties`?

**Comment by bastianvs** [comment]
I have started `./bin/kafka-monitor-start.sh config/kafka-monitor.properties `
and then 
started 

`./bin/end-to-end-test.sh --topic test --broker-list localhost:9092 --zookeeper localhost:2181 `

[2016-06-13 08:35:44,326] INFO kmf.services:name=end-to-end,type=produce-service:produce-availability-avg=0.0
kmf.services:name=end-to-end,type=produce-service:records-produced-total=0.0
kmf.services:name=end-to-end,type=consume-service:records-consumed-total=0.0
kmf.services:name=end-to-end,type=consume-service:records-lost-total=0.0
kmf.services:name=end-to-end,type=consume-service:records-duplicated-total=0.0
kmf.services:name=end-to-end,type=consume-service:records-delay-ms-avg=0.0
kmf.services:name=end-to-end,type=produce-service:records-produced-rate=0.0
kmf.services:name=end-to-end,type=produce-service:produce-error-rate=0.0
kmf.services:name=end-to-end,type=consume-service:consume-error-rate=0.0
 (com.linkedin.kmf.services.DefaultMetricsReporterService)

and not seeing anything in graphs. 

For the topic test produce and consume is working fine. 

producer:-

while true; do echo `date +"%s" | sha256sum` | sudo /home/y/libexec/server/kafka/bin/kafka-console-producer.sh --broker-list localhost:4443 --topic test; done

Consumer:-
 /home/y/libexec/server/kafka/bin/kafka-console-consumer.sh --zookeeper localhost:2181 --from-beginning --topic test

I can see all the stats using Kafa-manager but not seeing anything in kafka monitor looks like something is wrong.

In the UI http://<ip>:8000/index.html

No values in metrics table and metrics value 

Do i need to tune anything in configs ?

**Comment by bastianvs** [comment]
@lindong28 Seeing 404 while running HTTP Request. Do i need to change anything ?

```
$ curl localhost:8778/jolokia/read/kmf.services:type=produce-metrics,name=*/produce-availability-avg |json_pp 
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
101  2339  101  2339    0     0  1016k      0 --:--:-- --:--:-- --:--:-- 2284k
{
   "stacktrace" : "javax.management.InstanceNotFoundException: No MBean with pattern kmf.services:type=produce-metrics,name=* found for reading attributes\n\tat org.jolokia.handler.ReadHandler.searchMBeans(ReadHandler.java:160)\n\tat org.jolokia.handler.ReadHandler.fetchAttributesForMBeanPattern(ReadHandler.java:126)\n\tat org.jolokia.handler.ReadHandler.doHandleRequest(ReadHandler.java:116)\n\tat org.jolokia.handler.ReadHandler.doHandleRequest(ReadHandler.java:37)\n\tat org.jolokia.handler.JsonRequestHandler.handleRequest(JsonRequestHandler.java:161)\n\tat org.jolokia.backend.MBeanServerHandler.dispatchRequest(MBeanServerHandler.java:154)\n\tat org.jolokia.backend.LocalRequestDispatcher.dispatchRequest(LocalRequestDispatcher.java:99)\n\tat org.jolokia.backend.BackendManager.callRequestDispatcher(BackendManager.java:413)\n\tat org.jolokia.backend.BackendManager.handleRequest(BackendManager.java:158)\n\tat org.jolokia.http.HttpRequestHandler.executeRequest(HttpRequestHandler.java:197)\n\tat org.jolokia.http.HttpRequestHandler.handleGetRequest(HttpRequestHandler.java:86)\n\tat org.jolokia.jvmagent.handler.JolokiaHttpHandler.executeGetRequest(JolokiaHttpHandler.java:255)\n\tat org.jolokia.jvmagent.handler.JolokiaHttpHandler.doHandle(JolokiaHttpHandler.java:220)\n\tat org.jolokia.jvmagent.handler.JolokiaHttpHandler.handle(JolokiaHttpHandler.java:168)\n\tat com.sun.net.httpserver.Filter$Chain.doFilter(Filter.java:79)\n\tat sun.net.httpserver.AuthFilter.doFilter(AuthFilter.java:83)\n\tat com.sun.net.httpserver.Filter$Chain.doFilter(Filter.java:82)\n\tat sun.net.httpserver.ServerImpl$Exchange$LinkHandler.handle(ServerImpl.java:675)\n\tat com.sun.net.httpserver.Filter$Chain.doFilter(Filter.java:79)\n\tat sun.net.httpserver.ServerImpl$Exchange.run(ServerImpl.java:647)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)\n\tat java.lang.Thread.run(Thread.java:745)\n",
   "request" : {
      "attribute" : "produce-availability-avg",
      "mbean" : "kmf.services:name=*,type=produce-metrics",
      "type" : "read"
   },
   "status" : 404,
   "error" : "javax.management.InstanceNotFoundException : No MBean with pattern kmf.services:type=produce-metrics,name=* found for reading attributes",
   "error_type" : "javax.management.InstanceNotFoundException"
}
```

**Comment by lindong28** [comment]
@bastianvs Are you able to see any metrics printed in the console? Did you change zookeeper.connect and bootstrap.servers in the kafka-monitor.properties?

**Comment by nickwales** [comment]
I have a similar problem, everything works locally on my mac, on linux however the dashboards are visible but don't have any data. 

JMX endpoints work fine on all systems.

**Comment by lindong28** [comment]
@nickwales Sorry, we don't really have a full-time developer for the GUI. And the GUI has only been tested on Red Hat Enterprise Linux Workstation release 6. Which operating system did you run this GUI? Does it fail for all these systems?

The web GUI is result of a hack-day project and it is a simple quick-start GUI. We expect companies and advanced users to read metrics from JMX endpoints and write into their existing metrics display framework (e.g. Graphite). We real focus is on the Kafka Monitor core code and the JMX endpoints. We won't be able to provide sustained support for this GUI at this moment.

**Comment by anandsk123** [comment]
I am able to get the monitor running locally against a Kafka cluster by changing only the "topic" and "zookeeper.connect"

but when I run the monitor with the same config on a remote box it does not produce and send any msgs, please help with this.

**Comment by chocolateBlack** [comment]
@nickwales Can you access the url which is in the webapp/index.html, 
e.g https://ajax.googleapis.com/ajax/libs/jquery/1.12.0/jquery.min.js.
when you access to the http://hostname:8000/index.html, you can let the browser in the debug mode (F12 ) and then you will see what is going wrong in the GUI

**Comment by MercyPrasanna** [comment]
I had the same issue of UI not being populated when running on a docker container.

When I ran the debug mode on the browser as suggested, found that the web app is sending a request /jolokia on port 8778 which is not exposed outside the container. 

To solve this issue, I found the mapping port on the host for 8778 and then updated the index.html in kakfa-monitor webapp folder to query jolokia on the mapping port intead of 8778.

This worked for me and I am able to see the graphs now.

**Comment by Laxman-SM** [comment]
This also work for me. i am able to see graph UI. only updated jetty port from 8000 to 8001 to resolve conflict on my vagrant box.

**Comment by aliaksandrKachurka** [comment]
@bastianvs got the same problem. It appeared that kafka-monitor `kafka` and `kafka-clients` dependencies are incompatible with cluster version (0.10.x vs 0.9.x). Try to build kafka-monitor from 0.9.0.1 branch.

**Comment by lindong28** [comment]
@MercyPrasanna Thanks for figuring out the issue!
